lib/rules/selector-no-type/__tests__/index.js

```diff
@@ -165,6 +171,66 @@ testRule(rule, {
 
 testRule(rule, {
   ruleName,
+  config: [ true, { ignore: ["child"] } ],
+  skipBasicChecks: true,
+
+  accept: [ {
+    code: ".foo > div {}",
+    description: "class, child and type",
+  }, {
+    code: "#bar > div.foo {}",
+    description: "id, child and compounded",
+  }, {
+    code: ".foo, #bar > div {}",
+    description: "selector list with class, id, child and type",
+  } ],
+
+  reject: [ {
+    code: "div {}",
+    message: messages.rejected,
+  }, {
+    code: ".foo, div {}",
+    message: messages.rejected,
+  }, {
+    code: "div.foo {}",
+    message: messages.rejected,
+  }, {
+    code: ".foo div {}",
+    description: "descendant",
+    message: messages.rejected,
+  }, {
+    code: ".foo + div {}",
+    description: "next-sibling",
+    message: messages.rejected,
+  }, {
+    code: ".foo ~ div {}",
+    description: "following-sibling",
+    message: messages.rejected,
+  }, {
+    code: "div > foo {}",
+    description: "type and child type",
+  }, {
+    code: "div > a {}",
+    description: "type and child type",
+    message: messages.rejected,
```

Comment:
Good idea to add line and column here so it's clear which type selector is violating the rule.